sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxyUtil.java

```diff
@@ -0,0 +1,102 @@
+        if (options == null) {
+            return context;
+        }
+
+        Context optionsContext = options.getContext();
+        if (optionsContext != null && optionsContext != Context.NONE) {
+            context = CoreUtils.mergeContexts(context, optionsContext);
+        }
+
+        return context;
+    }
+
+    @SuppressWarnings("deprecation")
+    static void validateResumeOperationIsNotPresent(Method method, ClientLogger logger) {
+        // Use the fully-qualified class name as javac will throw deprecation warnings on imports when the class is
+        // marked as deprecated.
+        if (method.isAnnotationPresent(com.azure.core.annotation.ResumeOperation.class)) {
+            throw logger.logExceptionAsError(new IllegalStateException("'ResumeOperation' isn't supported."));
+        }
+    }
+
+    static Exception instantiateUnexpectedException(final UnexpectedExceptionInformation exception,
+                                                    final HttpResponse httpResponse, final byte[] responseContent, final Object responseDecodedContent) {
+        final int responseStatusCode = httpResponse.getStatusCode();
+        final String contentType = httpResponse.getHeaderValue("Content-Type");
+        final String bodyRepresentation;
+        if ("application/octet-stream".equalsIgnoreCase(contentType)) {
+            bodyRepresentation = "(" + httpResponse.getHeaderValue("Content-Length") + "-byte body)";
+        } else {
+            bodyRepresentation = responseContent == null || responseContent.length == 0
+                ? "(empty body)"
+                : "\"" + new String(responseContent, StandardCharsets.UTF_8) + "\"";
+        }
+
+        Exception result;
+        try {
+            final Constructor<? extends HttpResponseException> exceptionConstructor = exception.getExceptionType()
+                .getConstructor(String.class, HttpResponse.class, exception.getExceptionBodyType());
+            result = exceptionConstructor.newInstance("Status code " + responseStatusCode + ", " + bodyRepresentation,
+                httpResponse, responseDecodedContent);
+        } catch (ReflectiveOperationException e) {
+            String message = "Status code " + responseStatusCode + ", but an instance of "
+                + exception.getExceptionType().getCanonicalName() + " cannot be created."
+                + " Response body: " + bodyRepresentation;
+
+            result = new IOException(message, e);
+        }
+        return result;
+    }
+
+    /**
+     * Create an instance of the default serializer.
+     *
+     * @return the default serializer
+     */
+    static SerializerAdapter createDefaultSerializer() {
+        return JacksonAdapter.createDefaultSerializerAdapter();
+    }
+
+    /**
+     * Create the default HttpPipeline.
+     *
+     * @return the default HttpPipeline
+     */
+    static HttpPipeline createDefaultPipeline() {
```

Comment:
where is this used ?

Reply:
Used in static Rest Proxy create methods